File: palindromeNumber.py
class Solution:
    # @return a boolean
    def isPalindrome(self, x):
        if x < 0:
            return False
        reverse = 0
        original = x
        while x > 0:
            reverse = reverse * 10 + x % 10
            x /= 10
        return original == reverse

File: JAVA/palindromeNumber.java
public class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0)
            return false;
        int c = x, reverse = 0;
        while (c > 0){
            reverse = reverse * 10 + c % 10;
            c /= 10;
        }
        return x == reverse;
    }
}